### PR TITLE
Task/add query update action

### PR DESCRIPTION
**Pull request by AlvaroVega**
Issue https://github.com/telefonicaid/perseo-fe/issues/335
- [x] New action (NGSI query + update over result of query) using NGSI v2
  - [x] Refactor to update using batchUpdate
            https://conwetlab.github.io/ngsijs/stable/NGSI.Connection.html#.%22v2.batchUpdate%22__anchor
  - [x] Refactor to integrate into updateAction
- [x] Tested
- [x] New tests
- [x] Doc

native NGSIv2

example:
```
"action": {
      "type":"update",
      "parameters":{
          "filter": {
              "type": "Lamp"
          },
          "attributes": [
              {
                  "name":"status",
                  "type":"Text",
                  "value":"on"
              }
          ],
          "actionType": "UPDATE"
      }
}
```

### CODE REVIEW

File: documentation/plain_rules.md [hunk 1]
```diff
@@ -193,6 +194,7 @@ The `parameters` map includes the following fields:
 -   actionType: optional, type of CB action: APPEND or UPDATE. By default is APPEND.
 -   trust: optional, trust token for getting an access token from Auth Server which can be used to get to a Context
     Broker behind a PEP.
+-   filter: optional, a NGSI filter. If provided then updateAction is done over result of query.
```
Comment (fgalan) [inline documentation/plain_rules.md]: In addition, I'd suggest to modify the version field this way:

```
-   version: optional, The NGSI version for the update action. Set this attribute to `2` or `"2"` if you want to use
    NGSv2 format. `1` by default. However, if `filter` is used, this field is ignored and NGSIv2 is used.
```

File: documentation/plain_rules.md [hunk 2]
```diff
@@ -452,6 +452,32 @@ This attribute will take `null` as value.
...
+**Complete example using NGSv2 update action with filter in a rule:**
+
+```json
+{
+    "name": "blood_rule_update",
+    "text": "select *,\"blood_rule_update\" as ruleName, *, ev.BloodPressure? as Pressure from pattern [every ev=iotEvent(BloodPressure? > 1.5 and type=\"BloodMeter\")]",
+    "action": {
+        "type": "update",
+        "filter": {
+            "type": "SensorMetter"
```
Comment (fgalan) [inline documentation/plain_rules.md]: Maybe we cand add the following to the filter

"q": "status:on"

to illustrate how powerfull they are :)

File: test/unit/updateAction.js [hunk 3]
```diff
@@ -300,6 +299,13 @@ var expectedChanges = {
     }
 };
 
+var queryOptions = { type: 'AmbientLightSensor' };
+
+var expectedChanges2 = {
+    actionType: 'APPEND',
```
Comment (fgalan) [inline test/unit/updateAction.js]: Right syntax in NGSIv2 is camelCase for action types, i.e. "append".

Comment (AlvaroVega) [reply test/unit/updateAction.js]: fixed in d93078c1a87dfd9e54e00f3021a06c9893620b11

File: documentation/plain_rules.md (hunk 2 above)
Comment (AlvaroVega) [reply documentation/plain_rules.md]: fixed in d93078c1a87dfd9e54e00f3021a06c9893620b11

File: documentation/plain_rules.md (hunk 1 above)
Comment (AlvaroVega) [reply documentation/plain_rules.md]: fixed in d93078c1a87dfd9e54e00f3021a06c9893620b11

File: lib/models/updateAction.js [hunk 4]
```diff
@@ -135,7 +150,7 @@ function doRequest(action, event, token, callback) {
                 attributes: options.attributes
             }
         ],
-        updateAction: 'APPEND'
+        updateAction: 'append'
```
Comment (fgalan) [inline lib/models/updateAction.js]: Thid doRequest() is used by the NGSIv1 logic?

* In that case APPEND is correct
* If only used by NGSIv2 logic, then "append" is correct

Comment (AlvaroVega) [reply lib/models/updateAction.js]: fixed in a01067872a33567608238ec7683be42e35565137

**Pull request merged**